api/v1alpha1/etcd_types.go

```diff
@@ -133,8 +133,8 @@ type EtcdSpec struct {
 	// It must match the pod template's labels.
 	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
 	Selector *metav1.LabelSelector `json:"selector"`
-	// +optional
-	Labels map[string]string `json:"labels,omitempty"`
+	// +required
```

Comment:
This is required as the selector is mandatory. Labels cannot be made optional as a result. In fact, I will have to add a validating webhook to prevent creation of the resource if the selector and labels mismatch. This is present for sts in k8s. Ill adopt that logic here.

Reply:
👍 But as discussed offline, just making it as `required` doesn't prevent people from setting it to an empty map. That can only be addressed using a validating webhook which we can add separately in a different PR.